### PR TITLE
Custom preferred height of `Modal`

### PR DESCRIPTION
**Pull request by adamkudrna**
- Fix(web): Match the name and function of the expand-on-mobile toggle in `Modal`
- Feat(web): Make the preferred height and max height of `Modal` customizable #DS-723

### CODE REVIEW

File: packages/web/src/scss/components/Modal/_theme.scss [hunk 1]
```diff
@@ -18,10 +18,10 @@ $common-padding-x-tablet: tokens.$space-800;
 $dialog-width: 100%;
 $dialog-width-tablet: 640px;
 $dialog-width-desktop: 680px;
-$dialog-height: min-content;
-$dialog-height-tablet: min-content;
+$dialog-height: var(--modal-preferred-height-mobile, min-content);
+$dialog-height-tablet: var(--modal-preferred-height-tablet, min-content);
 $dialog-max-height: calc(100dvh - #{tokens.$space-1100});
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Modal/_theme.scss:23]: Maybe make mobile max height customizable too? But does it make sense? 🤔

**Review by literat — APPROVED** [review]
LGTM 🤷

**Review by tomassychra — APPROVED** [review]

File: packages/web/src/scss/components/Modal/README.md [hunk 2]
```diff
@@ -30,6 +30,49 @@ advantages.
...
+>
+  <!-- ModalDialog -->
+</dialog>
+```
+
+#### Custom Max Height
+
+The default maximum height of Modal is:
+
+- viewport height minus 64 px on mobile screens, and
```
Comment (crishpeen) [inline packages/web/src/scss/components/Modal/README.md:58]: It is actually not 64px, but `tokens.$space-600`. Which could be any value.

File: packages/web/src/scss/components/Modal/_ModalBody.scss [hunk 3]
```diff
@@ -9,20 +9,12 @@
     flex: 1 1 auto;
     min-height: map.get(theme.$typography, mobile, font-size) * map.get(theme.$typography, mobile, line-height); // 1.
     padding-inline: theme.$common-padding-x;
+    padding-block: theme.$body-padding-y;
```
Comment (crishpeen) [inline packages/web/src/scss/components/Modal/_ModalBody.scss:12]: This is BC and we want to implement this as a separate issue with a feature flag. See DS-763.

File: packages/web/src/scss/components/Modal/_theme.scss (hunk 1 above)
Comment (crishpeen) [reply packages/web/src/scss/components/Modal/_theme.scss:23]: I don't know. We can add it later when we find a use case for it.

File: packages/web/src/scss/components/Modal/README.md [hunk 4]
```diff
@@ -30,6 +30,49 @@ advantages.
...
+  id="example_1"
+  class="Modal Modal--composed"
+  aria-labelledby="example_1_title"
+  style="--modal-max-height-tablet: 700px"
+>
+  <!-- ModalDialog -->
+</dialog>
+```
+
+👉 Please note the max height on mobile screens is not customizable.
```
Comment (crishpeen) [inline packages/web/src/scss/components/Modal/README.md:74]: Maybe we could mention that if they would need it, they can let us know :)

**Review by crishpeen — CHANGES_REQUESTED** [review]

**Review by crishpeen — COMMENTED** [review]
Removing my blocking change request. GL.

File: packages/web/src/scss/components/Modal/index.html [hunk 5]
```diff
@@ -121,6 +121,174 @@ <h2 id="example_1_title" class="ModalHeader__title">Modal Title</h2>
...
+        <h2 id="example_3_title" class="ModalHeader__title">Modal with a Form</h2>
+        <button
+          type="button"
+          class="Button Button--tertiary Button--square Button--medium"
+          data-dismiss="modal"
+          data-target="#example_3"
+          aria-controls="example_3"
+          aria-expanded="false"
+        >
+          <svg class="Icon" width="24" height="24" aria-hidden="true">
```
Comment (crishpeen) [inline packages/web/src/scss/components/Modal/index.html:152]: There is no class Icon

**Review by crishpeen — APPROVED** [review]

File: packages/web/src/scss/components/Modal/_theme.scss (hunk 1 above)
Comment (adamkudrna) [reply packages/web/src/scss/components/Modal/_theme.scss:23]: OK.

File: packages/web/src/scss/components/Modal/README.md [hunk 6]
```diff
@@ -30,6 +30,49 @@ advantages.
...
+>
+  <!-- ModalDialog -->
+</dialog>
+```
+
+#### Custom Max Height
+
+The default maximum height of Modal is:
+
+- viewport height minus 1100 spacing on mobile screens, and
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web/src/scss/components/Modal/README.md:58]: Or describe this without numbers, e.g. use tokens instead?

[the remark this answers is not in the document]
Comment (crishpeen) [reply packages/web/src/scss/components/Modal/README.md:58]: This is actually a token value. It is space-1100 token :)

**Review by crishpeen — APPROVED** [review]

File: packages/web/src/scss/components/Modal/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (literat) [reply packages/web/src/scss/components/Modal/README.md:58]: Ok, then it is not understandable from that to me 🤷‍♂️

[the remark this answers is not in the document]
Comment (adamkudrna) [reply packages/web/src/scss/components/Modal/README.md:58]: Changed to the token name in Sass: `$spacing-1100`.

**Pull request merged**